### PR TITLE
Handled crash after unsuccesful load of libvlc on windows

### PR DESCRIPTION
**Pull request by devos50**

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm it seems that the tests are always failing because it cannot import `plyvel`?

**Comment by devos50** [comment]
retest this please

**Comment by synctext** [comment]
retest this please

**Comment by whirm** [comment]
@brussee deleted ~jenkins/.local when building the android stuff so plyvel went away. I've reinstalled it now.

**Comment by whirm** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

File: Tribler/vlc.py [hunk 1]
```diff
@@ -98,6 +100,15 @@ def bytes_to_str(b):
...
 _internal_guard = object()
 
+def load_libvlc_windows_dll():
+    try:
+        dll = ctypes.CDLL('libvlc.dll')
+    except WindowsError:
+        pass
+    finally:
+        # restore cwd after dll has been loaded
+        os.chdir(p)
```
Comment (whirm) [inline Tribler/vlc.py]: p doesn't exist here

File: Tribler/vlc.py [hunk 2]
```diff
@@ -98,6 +100,15 @@ def bytes_to_str(b):
 # instanciated.
 _internal_guard = object()
 
+def load_libvlc_windows_dll():
+    try:
+        dll = ctypes.CDLL('libvlc.dll')
```
Comment (whirm) [inline Tribler/vlc.py]: you are not returning the dll

File: Tribler/vlc.py [hunk 3]
```diff
@@ -98,6 +100,15 @@ def bytes_to_str(b):
 # instanciated.
 _internal_guard = object()
 
+def load_libvlc_windows_dll():
```
Comment (whirm) [inline Tribler/vlc.py]: This global method it's only used in one function, make it an inner method

File: Tribler/vlc.py [hunk 4]
```diff
@@ -98,6 +100,7 @@ def bytes_to_str(b):
 # instanciated.
 _internal_guard = object()
 
+
```
Comment (devos50) [inline Tribler/vlc.py]: This is to make PEP8 happy :+1:

File: Tribler/vlc.py [hunk 5]
```diff
@@ -48,6 +48,8 @@
 # Used by EventManager in override.py
 from inspect import getargspec
 
+from twisted.python.win32 import WindowsError
```
Comment (whirm) [inline Tribler/vlc.py]: I'm pretty sure this is a builtin exception, no need to import it

https://docs.python.org/2/library/exceptions.html#exceptions.WindowsError

**Comment by devos50** [comment]
@whirm confirmed to work when it can and cannot find VLC on Windows, will squash now.

**Comment by whirm** [comment]
Good, let's wait for `devel` to be stable and I'll merge this.

**Comment by whirm** [comment]
Merging as it passed by itself

**Pull request merged**